test/e2e/common/node/container_probe.go

```diff
@@ -512,6 +513,52 @@ var _ = SIGDescribe("Probing container", func() {
 		// 10s delay + 10s period + 5s grace period = 25s < 30s << pod-level timeout 500
 		RunLivenessTest(f, pod, 1, time.Second*30)
 	})
+
+	/*
+		Release: v1.23
+		Testname: Pod liveness probe, using grpc call, success
+		Description: A Pod is created with liveness probe on grpc service. Liveness probe on this endpoint will not fail. When liveness probe does not fail then the restart count MUST remain zero.
+	*/
+	ginkgo.It("should *not* be restarted with a GRPC liveness probe [NodeAlphaFeature:GRPCContainerProbe]", func() {
+		e2eskipper.SkipUnlessFeatureGateEnabled(kubefeatures.GRPCContainerProbe)
+
+		livenessProbe := &v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				GRPC: &v1.GRPCAction{
+					Port:    2379,
+					Service: nil,
+				},
+			},
+			InitialDelaySeconds: probeTestInitialDelaySeconds,
+			FailureThreshold:    1,
+		}
+
+		pod := gRPCServerPodSpec(nil, livenessProbe, "etcd")
+		RunLivenessTest(f, pod, 0, defaultObservationTimeout)
+	})
+
+	/*
+			Release: v1.23
+			Testname: Pod liveness probe, using grpc call, failure
+			Description: A Pod is created with liveness probe on grpc service. Liveness probe on this endpoint should fail because of wrong probe port.
```

Comment:
hm, description doesn't seem to match the test. Port is correct, service is specified explicitly though. So it looks like the port needs to be changed in code so this test will pass